### PR TITLE
Mock Battlefield

### PR DESCRIPTION
**Pull request by phukaokub**

### CODE REVIEW

**Comment by phukaokub** [comment]
This pull request introduces significant updates to the game’s state management and rendering system, particularly focusing on the initialization and selection states for a battle system. The most important changes include refactoring the state classes, adding new rendering methods, and creating new classes for cards, entities, and fields.

### State Management Updates:

* [`main.py`](diffhunk://#diff-b10564ab7d2c520cdd0243874879fb0a782862c3c902ab535faabe57d5a505e1L2-R2): Refactored the state initialization to include `BattleInitialState` and `BattleSelectState`, and added methods `RenderBackground` and `CreateBattleField` for rendering the HUD and battlefield. [[1]](diffhunk://#diff-b10564ab7d2c520cdd0243874879fb0a782862c3c902ab535faabe57d5a505e1L2-R2) [[2]](diffhunk://#diff-b10564ab7d2c520cdd0243874879fb0a782862c3c902ab535faabe57d5a505e1L16-R35) [[3]](diffhunk://#diff-b10564ab7d2c520cdd0243874879fb0a782862c3c902ab535faabe57d5a505e1R47-R48)
* [`src/dependency.py`](diffhunk://#diff-b85f827ef7bd92fb9a3016d55e99bff43180d3d0c1bdefc91fff421672b0322bR3-R10): Updated imports to use new battle states.
* [`src/states/BattleInitialState.py`](diffhunk://#diff-d1cca4475c5f4c1d9d2b6ace09fba01f196d1ba5a3d2c8aec7746929212dcacbR1-R77): Added a new state for initializing the battle, including methods for rendering cards and fields.
* [`src/states/BattleSelectState.py`](diffhunk://#diff-e0564c2b73a48f975e188e2dadebdb4bef6300674bec977dc96944f50cdd7428R1-R49): Added a new state for selecting actions during the battle, including methods for rendering cards, fields, and entities.

### Rendering and Entity Management:

* [`src/cardSystem/Card.py`](diffhunk://#diff-c7b085b059f336faa82322bafcbe26b3026aa7470005cdf546c348440a0e0834R1-R20): Introduced a `Card` class with methods for rendering and updating cards.
* [`src/cardSystem/Entity.py`](diffhunk://#diff-268e9765369047e84db57d06ac6cdbbfc886ccd99a79c2a17cd9fd9b3e4b9ee9R1-R34): Introduced an `Entity` class with methods for moving entities between fields and rendering them.
* [`src/cardSystem/Field.py`](diffhunk://#diff-6b948183acf326558c99094480757c1af751b16c1eec4f503f34417c090d3d84R1-R35): Introduced a `Field` class with methods for placing and removing entities and rendering fields.

### Constants and Cleanup:

* [`src/constants.py`](diffhunk://#diff-42d0f00c4780305311eae71fecca993a0523587cb4156e039cabd0d1ad8dbbb8R3-R11): Added new constants for card dimensions, HUD height, and field dimensions.
* Removed obsolete states and restructured state files for better organization. [[1]](diffhunk://#diff-aff02f215ffc94f33445b9438cfda5a2560e570764b4e59de9f5812be048ccb6L6-R6) [[2]](diffhunk://#diff-6d53f3f91f0dfae37a2311a2d07157f7be87c6df2378f64f27068188470cf4c0L6-R6) [[3]](diffhunk://#diff-bffdbfa3402cee65c1b9ca99885a3236870fa7e47f90dc1dcb455f695c446d25L1-L27) [[4]](diffhunk://#diff-4930cfe03b8f922c0e0ba9c421c2db21347b3aad7299da205e25f0c4e53fb027L1-L16)

File: src/cardSystem/Field.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import pygame
+from src.dependency import *
+
+class Field:
+    def __init__(self, index, position):
+        self.index = index  # Index in the row
+        self.position = position  # (x, y) coordinates for rendering
+        self.entity = None  # Store the entity occupying the field
+
+    def is_occupied(self):
```
Comment (phukaokub) [inline src/cardSystem/Field.py:10]: @rymTooo ตรงนี้กูงงว่ะเพื่อน อะไรงี้

วิธีการทำก็กดที่ File change -> จิ้มบรรทัดที่สนใจจะมีปุ่ม + -> พิมคอมเม้นไว้

**Comment by rymTooo** [comment]
where is the comment??

File: src/cardSystem/Field.py (hunk 1 above)
Comment (rymTooo) [reply src/cardSystem/Field.py:10]: I don't pim thai in pc na. I can't a hahahah

Comment (phukaokub) [reply src/cardSystem/Field.py:10]: We implemented it on entity class and the plac_entity function below

**Pull request merged**